Library/Homebrew/test/inreplace_spec.rb

```diff
@@ -0,0 +1,253 @@
+    end
+
+    context "shell-style variable" do
+      let(:string) do
+        <<-EOS.undent
+          OTHER=def
+          FLAG=abc
+          FLAG2=abc
+        EOS
+      end
+
+      it "is successfully replaced" do
+        subject.change_make_var! "FLAG", "def"
+        expect(subject).to eq <<-EOS.undent
+          OTHER=def
+          FLAG=def
+          FLAG2=abc
+        EOS
+      end
+    end
+  end
+
+  describe "#remove_make_var!" do
+    context "flag" do
+      context "with spaces" do
+        let(:string) do
+          <<-EOS.undent
+            OTHER=def
+            FLAG = abc
+            FLAG2 = def
+          EOS
+        end
+
+        it "is successfully removed" do
+          subject.remove_make_var! "FLAG"
+          expect(subject).to eq <<-EOS.undent
+            OTHER=def
+            FLAG2 = def
+          EOS
+        end
+      end
+
+      context "with tabs" do
+        let(:string) do
+          <<-EOS.undent
+            CFLAGS\t=\t-Wall -O2
+            LDFLAGS\t=\t-lcrypto -lssl
+          EOS
+        end
+
+        it "is successfully removed" do
+          subject.remove_make_var! "LDFLAGS"
+          expect(subject).to eq <<-EOS.undent
+            CFLAGS\t=\t-Wall -O2
+          EOS
+        end
+      end
+    end
+
+    context "multiple flags" do
+      let(:string) do
+        <<-EOS.undent
+          OTHER=def
+          FLAG = abc
+          FLAG2 = def
+          OTHER2=def
+        EOS
+      end
+
+      specify "are be successfully removed" do
+        subject.remove_make_var! ["FLAG", "FLAG2"]
+        expect(subject).to eq <<-EOS.undent
+          OTHER=def
+          OTHER2=def
+        EOS
+      end
+    end
+  end
+
+  describe "#get_make_var" do
+    context "with spaces" do
+      let(:string) do
+        <<-EOS.undent
+          CFLAGS = -Wall -O2
+          LDFLAGS = -lcrypto -lssl
+        EOS
+      end
+
+      it "extracts the value for a given variable" do
+        expect(subject.get_make_var("CFLAGS")).to eq("-Wall -O2")
+      end
+    end
+
+    context "with tabs" do
+      let(:string) do
+        <<-EOS.undent
+          CFLAGS\t=\t-Wall -O2
+          LDFLAGS\t=\t-lcrypto -lssl
+        EOS
+      end
+
+      it "extracts the value for a given variable" do
+        expect(subject.get_make_var("CFLAGS")).to eq("-Wall -O2")
+      end
+    end
+  end
+
+  describe "#sub!" do
+    let(:string) { "foo" }
+
+    it "replaces the first occurence" do
+      subject.sub!("o", "e")
+      expect(subject).to eq("feo")
+    end
+  end
+
+  describe "#gsub!" do
+    let(:string) { "foo" }
+
+    it "replaces the all occurences" do
+      subject.gsub!("o", "e") # rubocop:disable Performance/StringReplacement
+      expect(subject).to eq("fee")
+    end
+  end
+end
+
+describe Utils::Inreplace do
+  let(:file) { Tempfile.new("test") }
+
+  before(:each) do
+    file.write <<-EOS.undent
+      a
+      b
+      c
+    EOS
+  end
+
+  after(:each) do
+    file.unlink
+  end
```

Comment:
I'd like to always use `do … end` for `before` and `after` hooks, since they *do* something.